src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -817,153 +817,106 @@ public override bool Equals(Object obj)
         public bool Equals(String value)
         {
             if (this == null)                        // this is necessary to guard against reverse-pinvokes and
-                throw new NullReferenceException();  // other callers who do not use the callvirt instruction
+                ThrowHelper.ThrowNullReferenceException();  // other callers who do not use the callvirt instruction
 
-            if (object.ReferenceEquals(this, value))
-                return true;
-
-            // NOTE: No need to worry about casting to object here.
-            // If either side of an == comparison between strings
-            // is null, Roslyn generates a simple ceq instruction
-            // instead of calling string.op_Equality.
-            if (value == null)
-                return false;
-            
-            if (this.Length != value.Length)
-                return false;
+            return EqualsOrdinalCaseSensitive(this, value);
```

Comment:
and then `Equals(String a, String b)`  would be the only caller for `EqualsHelperComparison` so the latter code could get inlined into the former again making code clearer and should have no effect on the output.